### ISSUE TITLE
Suggesting Other Possibly Useful Widgets

### ISSUE THREAD

**Issue by Wajinn** [malisipi/mui#10]
Suggesting to add some well known widgets that seem to be missing or would be nice to add, to the existing awesome collection:

1) Combo box (similar to existing select box, but all options always viewable)
2) List box (similar to select box, but all options viewable, and clicking selection is direct action)
3) UpDown box (has 2 arrows on right side, where user can click up or down, to increase or decrease a displayed value)
4) List view (kind of similar to your [cell example](https://github.com/malisipi/mui/blob/main/examples/cells.v), but as a configurable widget)

**Comment by malisipi** [comment]
I agree the controls are required. But i could not understand a few things,
1- How is the combo box different from the select box? Is it always viewable or viewable until user-select?
2- Should the list view be editable? If not, what is different between the table widget?
Also thank you for your contribution.

**Comment by Wajinn** [comment]
Thanks for the response.  You are doing a good job.

> 1- How is the combo box different from the select box? 

Sometimes you don't want to hide options from users, and don't mind the space it uses.  It can be faster or more convenient for the user to see all possible options.  

> Is it always viewable or viewable until user-select?

All combo box contents are viewable, unless so long that a scroll bar is used, to stay within certain spacial limits.  Where with the select box, contents are hidden, until clicked upon.

 > 2- Should the list view be editable?

Usually, no.  List views are not editable.  A similar widget that would be editable is called a DataGrid view.

>  If not, what is different between the table widget? 

A list view usually has rows, with no columns.  And usually the entire "row" is highlighted, before selection.

**Comment by malisipi** [comment]
I have implemented list view with [e41036d](https://github.com/malisipi/mui/commit/e41036d04daf67023dfecfb9c9e7345a563c7813) commit. It has accessibility issues with keyboard for now. It will be fixed in next commits.
If you want to see a working example, you can browse [demo_with_list.v](https://github.com/malisipi/mui/blob/main/examples/demo_with_list.v) in examples.

**Comment by malisipi** [comment]
1. You can insert rows and row height will be `total height` / `number of rows`. Automatic scrollbars is not implemented yet on any widget and I'm not sure how to implement that. (I could misunderstand that. If it, please correct me).
2. Text size is really needed property. I will implement on any widget require that. ( Also that is related with #9 )

Also UpDown widget (aka Spinner) is finished with accessibility support.

![Spinner](https://user-images.githubusercontent.com/87281783/213930134-f70bc01b-0f7d-476f-b077-5af33c00a933.png)
[Source Code of Example](https://github.com/malisipi/mui/blob/main/examples/spinner.v)

**Comment by malisipi** [comment]
I'm working on making easier to use scrollbar widget.
It will be easy to integrate with other widgets.
```v
app.list(id:"list", table:user_list x:300, y:220, width:380, height:100, row_height:40)
app.scrollbar(id:"list_scrollbar", x:680, y:220, width: 20, height:100, vertical:true, connected_widget:app.get_object_by_id("list")[0])
```
It's limited by vertical scrollbars for now. I will support horizontal scrollbars. Also i will support another widgets (includes frames). And i will not make a separate scroll view. Because it's same with frame widgets that has scrollbars. However, i will write a macro function to make that.

You can watch the video to see how works. And you can follow the progress from [extended-scrollbar-support](https://github.com/malisipi/mui/tree/extended-scrollbar-support) branch. (It's not merged with main branch yet.)

PS: Also I supported `row_height` in tables and lists.

[scroll_list.webm](https://user-images.githubusercontent.com/87281783/216105026-68eee2b9-6b35-442a-935b-b6d29e5223c4.webm)
